ENIGMAsystem/SHELL/Universal_System/Extensions/Json/short_json_converter.h

```diff
@@ -0,0 +1,213 @@
+/**
+ *  @file short_json_converter.cpp
+ *  @section License
+ *
+ *      Copyright (C) 2023 Saif Kandil
+ *
+ *      This file is a part of the ENIGMA Development Environment.
+ *
+ *      This program is free software: you can redistribute it and/or modify
+ *      it under the terms of the GNU General Public License as published by
+ *      the Free Software Foundation, either version 3 of the License, or
+ *      (at your option) any later version.
+ *
+ *      This program is distributed in the hope that it will be useful,
+ *      but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *      MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *      GNU General Public License for more details.
+ *
+ *      You should have received a copy of the GNU General Public License
+ *      along with this program.  If not, see <http://www.gnu.org/licenses/>.
+**/
+
+#ifndef SHORT_JSON_CONVERTER_H
+#define SHORT_JSON_CONVERTER_H
+
+#include <iostream>
+#include <queue>
+#include <stack>
+#include <string>
+
+namespace enigma {
+
+class ShortJSONConverter {
+ private:
+  /********************************************
+    *
+    *    Error reporting system.
+    * 
+    *    TODO: implement this system.
+    *
+    *********************************************/
+
+  std::string format_human_readable_error_message(std::string& message);
+
+  /********************************************
+    *
+    *    Main system.
+    * 
+    *    '[' means start of a new level, and ']' is the end of that new level.
+    * 
+    *    New levels are considered "array of objects" by default. If a "key/value pair" is found
+    *    in current level then state of current level is changed.
+    * 
+    *           "array of objects" ----> single JSON "object"
+    *
+    *********************************************/
+  enum levelState { arrayOfObjectsState, objectState };
```

Comment:
You should use `enum class` to make it obvious that the names are enumerators, as it can be confusing when you use the same naming scheme as variables.